rm-main/src/ui/tabs/torrents/mod.rs

```diff
@@ -120,11 +120,17 @@ impl TorrentsTab {
             .general
             .accent_color);
 
-        let table_widget = Table::new(torrent_rows, table_manager_lock.widths)
-            .header(Row::new(
-                table_manager_lock.header().iter().map(|s| s.as_str()),
-            ))
-            .highlight_style(highlight_table_style);
+        let table_widget: Table;
```

Comment:
Ratatui's widgets use lite builder pattern so you can write:
```rust
let table_widget = Table::new(torrent_rows, table_manager_lock.widths)  
                           .highlight_style(highlight_table_style)
```
and then decide whether to add the headers in the ifs, in order to not repeat code

Reply:
<img width="1412" alt="Screenshot 2024-06-23 at 6 02 52 PM" src="https://github.com/intuis/rustmission/assets/26812563/5b186de1-add9-4e93-be00-de9fa35c68e4">

not really sure how to deal with this error? the only way i was able to fix it was with this way of doing if-else .. what would u recommend?

Reply:
<img width="713" alt="Screenshot 2024-06-23 at 6 05 13 PM" src="https://github.com/intuis/rustmission/assets/26812563/a53aed58-12c2-4515-b868-9fa34b097be2">

this works but i really didn't want to clone .. is this a better alternative?

Reply:
This should work:
```rust
let table_widget = {
  let table = Table::new(...).highlight_style(...);
  if hide_headers {
     table.header(...)
  } else {
    table
 }
}
```